jenkins/jdbc-stage-maven-release.jenkinsfile

```diff
@@ -0,0 +1,80 @@
+lib = library(identifier: 'jenkins@main', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/opensearch-project/opensearch-build.git',
+]))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-AL2-X64-C54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ci-runner-centos7-opensearch-build-v2'
+            args '-e JAVA_HOME=/opt/java/openjdk-11'
+            alwaysPull true
+        }
+    }
+    triggers {
+        GenericTrigger(
+            genericVariables: [
+                [key: 'ref', value: '$.ref'],
+                [key: 'VERSION', value: '$.ref', regexpFilter: 'refs/tags/v' ],
+            ],
+            tokenCredentialId: 'jenkins-sql-generic-webhook-token',
+            causeString: 'A tag was cut on opensearch-project/sql repository causing this workflow to run',
+            printContributedVariables: false,
+            printPostContent: false,
+            regexpFilterText: '$ref',
+            regexpFilterExpression: '^refs/tags/.*'
+        )
+    }
+    environment {
+        ARTIFACT_PATH = "$WORKSPACE/build/repository/org/opensearch/client/sql/$VERSION"
+    }
+    stages {
+        stage('Publish to Maven Local') {
+            steps {
+                // checkout the commit
+                checkout([
+                    $class: 'GitSCM', userRemoteConfigs: [[url: 'https://github.com/opensearch-project/sql.git']],
+                    branches: [[name: "$ref"]]
+                        ])
+
+                dir('sql-jdbc') {
+                    // publish maven artifacts
+                    sh('./gradlew --no-daemon publishPublishMavenPublicationToLocalRepoRepository')
+                }
+            }
+        }
```

Comment:
Will need to discuss a bit as we want two options: triggered by tag or manually trigger on Jenkins.
Just in case there is issues with github and we can still release.
See maps server jenkinsfile for more references:
https://github.com/peterzhuamazon/maps/blob/a40788a611e1d0f573d715e0b1ce59b7fb33e4d0/jenkins/maps-server-release.jenkinsfile